massa-event-cache/src/event_cache.rs

```diff
@@ -0,0 +1,460 @@
+        }
+
+        let mut batch = WriteBatch::default();
+        for event in events {
+            let event_key = {
+                let mut event_key = event.context.slot.to_bytes_key().to_vec();
+                event_key.extend(event.context.index_in_slot.to_be_bytes());
+                event_key
+            };
+            let mut event_buffer = Vec::new();
+            self.event_ser.serialize(&event, &mut event_buffer).unwrap();
+
+            batch.put(event_key, event_buffer);
+        }
+        self.db.write(batch).expect(CRUD_ERROR);
+        self.entry_count = self.entry_count.saturating_add(events_len);
+
+        debug!("(Events insert) entry_count is: {}", self.entry_count);
+    }
+
+    fn db_iter(&self, mode: Option<IteratorMode>) -> DBIterator {
+        self.db.iterator(mode.unwrap_or(IteratorMode::Start))
+    }
+
+    pub(crate) fn get_filtered_sc_output_events<'b, 'a: 'b>(
+        &'a self,
+        filter: &'b EventFilter,
+    ) -> impl Iterator<Item = SCOutputEvent> + 'b {
+        self.db_iter(Some(IteratorMode::Start)).filter_map(|kvb| {
+            let kvb = kvb.unwrap();
+            let (_rem, event) = self
+                .event_deser
+                .deserialize::<DeserializeError>(kvb.1.as_bytes())
+                .expect(EVENT_DESER_ERROR);
+
+            if let Some(start) = filter.start {
+                if event.context.slot < start {
+                    return None;
+                }
+            }
+            if let Some(end) = filter.end {
+                if event.context.slot >= end {
+                    return None;
+                }
+            }
+            if let Some(is_final) = filter.is_final {
+                if event.context.is_final != is_final {
+                    return None;
+                }
+            }
+            if let Some(is_error) = filter.is_error {
+                if event.context.is_error != is_error {
+                    return None;
+                }
+            }
+            match (
+                filter.original_caller_address,
+                event.context.call_stack.front(),
+            ) {
+                (Some(addr1), Some(addr2)) if addr1 != *addr2 => return None,
+                (Some(_), None) => return None,
+                _ => (),
+            }
+            match (filter.emitter_address, event.context.call_stack.back()) {
+                (Some(addr1), Some(addr2)) if addr1 != *addr2 => return None,
+                (Some(_), None) => return None,
+                _ => (),
+            }
+            match (
+                filter.original_operation_id,
+                event.context.origin_operation_id,
+            ) {
+                (Some(addr1), Some(addr2)) if addr1 != addr2 => return None,
+                (Some(_), None) => return None,
+                _ => (),
+            }
+            Some(event)
+        })
+    }
+
+    /// Try to remove as much as `self.amount_to_snip` entries from the db
```

Comment:
much -> many ?

that being said the comment does not in phase with the function implementation, (if snip_amount > self.amount_to_snip) more than self_amount_to_snip are removed
so I guess

```suggestion
    /// Try to remove at least `self.amount_to_snip` entries from the db
```

(and from the call above
```rust
if self.entry_count + events_len >= self.max_entry_count {
    let snip_amount = max(self.snip_amount, events_len);
    self.snip(Some(snip_amount));
}
```
)

In both cases, I don't feel the function enforce any boundary limitation. Maybe the comment is just wrong ^^

Reply:
Fixed the comment. Let me know if this is clearer now?